librosa/util/_nnls.py

```diff
@@ -146,17 +164,13 @@ def nnls(A, B, **kwargs):
     n_columns = MAX_MEM_BLOCK // (A.shape[-1] * A.itemsize)
     n_columns = max(n_columns, 1)
 
-    # Process in blocks:
-    if B.shape[-1] <= n_columns:
-        return _nnls_lbfgs_block(A, B, **kwargs).astype(A.dtype)
-
-    x = np.linalg.lstsq(A, B, rcond=None)[0].astype(A.dtype)
-    np.clip(x, 0, None, out=x)
-    x_init = x
-
-    for bl_s in range(0, x.shape[-1], n_columns):
-        bl_t = min(bl_s + n_columns, B.shape[-1])
-        x[:, bl_s:bl_t] = _nnls_lbfgs_block(
-            A, B[:, bl_s:bl_t], x_init=x_init[:, bl_s:bl_t], **kwargs
-        )
-    return x
+    # if single channel
+    if B.ndim == 2:
+        x = _process_in_blocks(A,B,n_columns,**kwargs)    
+    elif B.ndim == 3:
+        x = []
+        for iB in B:
+            x.append(_process_in_blocks(A,iB,n_columns,**kwargs))
+    else:
+        raise ParameterError("Target matrix B should have ndim <= 3 but has B.ndim: {:d}".format(B.ndim))
+    return np.array(x)
```

Comment:
Please revert this change.  We should not restrict to ndim<3; if it doesn't work using the reshape operations from before, then something is wrong.